staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -82,57 +82,88 @@ type kmsv2PluginProbe struct {
 	l            *sync.Mutex
 }
 
+type kmsHealthChecker []healthz.HealthChecker
+
+func (k kmsHealthChecker) Name() string {
+	return "kms-providers"
+}
+
+func (k kmsHealthChecker) Check(req *http.Request) error {
+	for i := range k {
+		checker := k[i]
+		if err := checker.Check(req); err != nil {
+			return fmt.Errorf("%s healthz check failed: %w", checker.Name(), err)
```

Comment:
also, make it easy to detect the failing name, don't need to include "healthz check failed", that's already clear from context or the wrapper the /healthz endpoint adds as a prefix

```suggestion
			return fmt.Errorf("%s: %w", checker.Name(), err)
```